### ISSUE TITLE
xinclude 1.1 attribute copying is not working

### ISSUE THREAD

**Issue by dwcramer** [ndw/xmlcalabash1#138]
Update: This is a documentation bug. You must add cx:copy-attributes="true" for this feature to work.

The xinclude 1.1 spec provides for a mechanism for copying attributes in a foreign namespace from the xi:include element into the result doc: 

http://www.w3.org/TR/2012/WD-xinclude-11-20121009/#attribute-copying

I understand this feature has already been implemented in Calabash:
https://github.com/ndw/xmlcalabash1/commit/84996dc1cfa3bcfd013af24963a86770386e073f

However, given the following pipeline and using Calabash 1.0.16:

```
<p:pipeline xmlns:p="http://www.w3.org/ns/xproc" version="1.0">
  <p:xinclude fixup-xml-base="true" name="xincluding"/>
</p:pipeline>
```

When I process the xinclude 1.1 test suite for attribute copying, the eg:root attributes are not copied into the result document as expected:

```
dcramer@anatine ~/Desktop/xinclude-11-test
$ calabash -i
source=http://www.w3.org/XML/2013/10/xinclude-11-test-suite/spec/attcopy-1.xml
test.xpl
<document xmlns:xi="http://www.w3.org/2001/XInclude"
xmlns:eg="http://example.org/namespace/example">
<p>This example includes a “definition” paragraph from some document
twice using attribute copying.</p>

<para xml:id="def"
xml:base="http://www.w3.org/XML/2013/10/xinclude-11-test-suite/spec/src.xml">Some
definition.</para>

<para xml:id="def"
xml:base="http://www.w3.org/XML/2013/10/xinclude-11-test-suite/spec/src.xml">Some
definition.</para>

</document>
```

**Comment by innovimax** [comment]
I fail to see where in the spec, it says that attribute are copied? Can you point us to the relevant part of the spec?

http://www.w3.org/TR/xinclude/#include_element

**Comment by dwcramer** [comment]
This is a new feature in xinclude 1.1:
http://www.w3.org/TR/2012/WD-xinclude-11-20121009/#attribute-copying
I notice this commit is related to the feature:
https://github.com/ndw/xmlcalabash1/commit/84996dc1cfa3bcfd013af24963a86770386e073f

**Comment by dwcramer** [comment]
Ok, this is actually a bug resulting from the fact that this feature is undocumented. I took a look at the source and found that you have to put cx:copy-attributes="true" on the p:xinclude.

**Comment by ndw** [comment]
Right. I still need to clean this up and make it really conformant, so don't close the bug :-)

**Comment by dwcramer** [comment]
@ndw: How will the behavior change once it's cleaned up? I expect that the need to add cx:copy-attributes will go away. Would the cx:root stuff change/go away as well?

**Comment by ndw** [comment]
Yes copy-attributes and root both go away.

**Comment by ndw** [comment]
Fixed in 1.0.20, I believe
